monero/wallet_client.go

```diff
@@ -36,37 +36,29 @@ const (
 
 // WalletClient represents a monero-wallet-rpc client.
 type WalletClient interface {
-	Lock()
-	Unlock()
 	GetAccounts() (*wallet.GetAccountsResponse, error)
 	GetAddress(idx uint64) (*wallet.GetAddressResponse, error)
-	PrimaryWalletAddress() mcrypto.Address
+	PrimaryAddress() mcrypto.Address
 	GetBalance(idx uint64) (*wallet.GetBalanceResponse, error)
-	Transfer(to mcrypto.Address, accountIdx uint64, amount *coins.PiconeroAmount) (*wallet.TransferResponse, error)
-	SweepAll(to mcrypto.Address, accountIdx uint64) (*wallet.SweepAllResponse, error)
-	WaitForReceipt(req *WaitForReceiptRequest) (*wallet.Transfer, error)
-	GenerateFromKeys(
-		kp *mcrypto.PrivateKeyPair,
-		restoreHeight uint64,
-		filename,
-		password string,
-		env common.Environment,
-	) error
-	GenerateViewOnlyWalletFromKeys(
-		vk *mcrypto.PrivateViewKey,
-		address mcrypto.Address,
-		restoreHeight uint64,
-		filename,
-		password string,
-	) error
+	Transfer(
+		ctx context.Context,
+		to mcrypto.Address,
+		accountIdx uint64,
+		amount *coins.PiconeroAmount,
+		numConfirmations uint64,
+	) (*wallet.Transfer, error)
+	SweepAll(
+		ctx context.Context,
+		to mcrypto.Address,
+		accountIdx uint64,
+		numConfirmations uint64,
+	) ([]*wallet.Transfer, error)
+	CreateABWalletConf(walletNamePrefix string) *WalletClientConf
+	WalletName() string
 	GetHeight() (uint64, error)
-	GetChainHeight() (uint64, error)
-	Refresh() error
-	CreateWallet(filename, password string) error
-	OpenWallet(filename, password string) error
-	CloseWallet() error
```

Comment:
`Lock`, `Unlock`, `Refresh`, `OpenWallet`, `CloseWallet`, `CreateWallet`, `GetChainHeight`, and `WaitForReceipt` were all removed from this package external interface.